### PR TITLE
👌 IMP: Optimizer refactor and Hogwild trainers

### PR DESCRIPTION
**Pull request by ianagbip1oti**
## Summary by CodeRabbit

* **New Features**
  * Training now uses a unified AdamW optimization flow across policy and value networks.
  * Training runs include expanded parameter update schedules and improved parallel gradient aggregation.
  * Neural networks now expose a consistent parameter representation for optimization.

* **Improvements**
  * Policy training uses adjusted target weighting.
  * Training progress displays loss values with greater precision.

* **Removed**
  * Learning-rate analysis reports and exports are no longer generated.

### CODE REVIEW

File: crates/princhess-train/src/bin/train-eg-policy.rs [hunk 1]
```diff
@@ -841,40 +797,42 @@ fn gradients_batch(
...
-    thread::scope(|s| {
+    let metrics: Vec<BatchMetrics> = thread::scope(|s| {
         batch
             .chunks(size)
-            .zip(thread_buffers.iter_mut())
-            .zip(thread_metrics.iter_mut())
-            .for_each(|((chunk, inner_gradients), inner_metrics)| {
+            .map(|chunk| {
                 s.spawn(move || {
+                    let grads = unsafe { &mut *(gradients_ptr as *mut EgPolicyNetwork) };
```
[the remark this answers is not in the document]
Comment (ianagbip1oti) [reply crates/princhess-train/src/bin/train-eg-policy.rs:816]: This was discussed on #493. For now we accept the undefined behavior, and save a fix as later (btw do you think it would be better if *mut XxxPolicyNetwork was used instead?
